src/Kunstmaan/UserManagementBundle/Controller/UsersController.php

```diff
@@ -197,57 +196,55 @@ public function editAction(Request $request, $id)
                 return new RedirectResponse(
                     $this->generateUrl(
                         'KunstmaanUserManagementBundle_settings_users_edit',
-                        array('id' => $id)
+                        ['id' => $id]
                     )
                 );
             }
         }
 
-        $params = array(
+        $params = [
             'form' => $form->createView(),
             'user' => $user,
-        );
+        ];
 
         if ($tabPane) {
-            $params = array_merge($params, array('tabPane' => $tabPane));
+            $params = array_merge($params, ['tabPane' => $tabPane]);
         }
 
         return $params;
     }
 
     /**
-     * Delete a user
-     *
-     * @param Request $request
-     * @param int     $id
-     *
-     * @Route("/{id}/delete", requirements={"id" = "\d+"}, name="KunstmaanUserManagementBundle_settings_users_delete", methods={"POST"})
+     * @Route("/delete", requirements={"id" = "\d+"}, name="KunstmaanUserManagementBundle_settings_users_delete", methods={"POST"})
      *
-     * @throws AccessDeniedException
-     *
-     * @return array
+     * @throws Exception
      */
-    public function deleteAction(Request $request, $id)
+    public function deleteAction(Request $request)
     {
         $this->denyAccessUnlessGranted('ROLE_SUPER_ADMIN');
 
         /* @var EntityManager $em */
         $em = $this->getDoctrine()->getManager();
-        /* @var UserInterface $user */
-        $user = $em->getRepository($this->container->getParameter('fos_user.model.user.class'))->find($id);
-        if (!\is_null($user)) {
-            $userEvent = new UserEvent($user, $request);
-            $this->container->get('event_dispatcher')->dispatch(UserEvents::USER_DELETE_INITIALIZE, $userEvent);
-
-            $em->remove($user);
-            $em->flush();
-
-            $this->addFlash(
-                FlashTypes::SUCCESS,
-                $this->container->get('translator')->trans('kuma_user.users.delete.flash.success.%username%', [
-                    '%username%' => $user->getUsername(),
-                ])
-            );
+        $user = new User();
```

Comment:
Can we do this? I think we need to use the userManager to get a new instance or inject the correct parameter because we let the user override the user class

Reply:
Its only for storing the ID in the form. The correct user type is later gotten from the repository with the ID stored in this entity. I will refactor it anyway to use the correct class from the beginning